src/lib/construct/staticSite.ts

```diff
@@ -130,6 +133,27 @@ export class StaticSite extends CommonConstruct {
     this.siteOrigin = new origins.S3Origin(this.siteBucket)
   }
 
+  /**
+   * @summary Method to create a site cloudfront function
+   * @protected
+   */
+  protected createSiteCloudfrontFunction() {
+    if (this.props.siteCloudfrontFunctionProps) {
+      this.siteCloudfrontFunction = this.cloudFrontManager.createCloudfrontFunction(
+        `${this.id}-function`,
+        this,
+        this.props.siteCloudfrontFunctionProps
+      )
+
+      this.siteFunctionAssociations = [
```

Comment:
Can you split this association into it's own method? That way more associations can be done by overriding just that method